src/components/card_item/CardItem.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { NavLink } from 'react-router-dom';
+import { toggleFavorite } from '../../store/features/favorites/favorites_slice';
+
+import type { RootState } from '../../store';
+import { type Post } from '../../models/post.interface';
+
+interface CardItemProps {
+  post: Post;
+}
+
+// Custom hook for favorites logic
+const useFavorites = (post: Post) => {
+  const dispatch = useDispatch();
+  const favorites = useSelector(
+    (state: RootState) => state.favorites.favorites
+  );
+
+  const isFavorite = useMemo(
+    () => favorites.some((fav) => fav.id === post.id),
+    [favorites, post.id]
+  );
+
+  const handleToggleFavorite = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      e.stopPropagation();
+      dispatch(toggleFavorite(post));
+    },
+    [dispatch, post]
+  );
+
+  return { isFavorite, handleToggleFavorite };
+};
```

Comment:
could you create a separate file for this. Otherwise the file will grow which is difficult to read.

Reply:
Also this hooks needs to be tested